extensions/oplogPopulator/modules/ConnectorsManager.js

```diff
@@ -105,40 +105,24 @@ class ConnectorsManager {
 
     /**
      * Creates a connector
-     * @param {boolean} spawn should connector be spawned
-     * @returns {Promise|Connector} created connector
-     * @throws {InternalError}
+     * @returns {Connector} created connector
      */
-    async addConnector(spawn = true) {
-        try {
-            // generate connector name
-            const connectorName = this._generateConnectorName();
-            // get connector config
-            const config = this._getDefaultConnectorConfiguration(connectorName);
-            // initialize connector
-            const connector = new Connector({
-                name: connectorName,
-                config,
-                buckets: [],
-                logger: this._logger,
-                kafkaConnectHost: this._kafkaConnectHost,
-                kafkaConnectPort: this._kafkaConnectPort,
-            });
-            if (spawn) {
-                await connector.spawn();
-            }
-            this._logger.debug('Successfully created connector', {
-                method: 'ConnectorsManager.addConnector',
-                connector: connector.name
-            });
-            return connector;
-        } catch (err) {
-            this._logger.error('An error occurred while creating connector', {
-                method: 'ConnectorsManager.addConnector',
-                error: err.description || err.message,
-            });
-            throw errors.InternalError.customizeDescription(err.description);
-        }
+    addConnector() {
+        // generate connector name
+        const connectorName = this._generateConnectorName();
+        // get connector config
+        const config = this._getDefaultConnectorConfiguration(connectorName);
+        // initialize connector
+        const connector = new Connector({
+            name: connectorName,
+            config,
+            buckets: [],
+            isRunning: false,
+            logger: this._logger,
+            kafkaConnectHost: this._kafkaConnectHost,
+            kafkaConnectPort: this._kafkaConnectPort,
+        });
+        return connector;
```

Comment:
not sure how to handle `onConnectorsInstantiated` metrics, now that the connectors are not created on startup :-/
it seems to me we are interested in the number of connectors created & destroyed (as 2 counters, when we actually creaste/destroy one connector), but not sure how it relates to this one.